scilpy/tractograms/intersection_finder.py

```diff
@@ -0,0 +1,220 @@
+import time
+import math
+import logging
+import numpy as np
+
+from scipy.spatial import KDTree
+from scilpy.tracking.fibertube import (segment_tractogram,
+                                       dist_segment_segment)
+from dipy.io.stateful_tractogram import StatefulTractogram
+from scilpy.io.utils import v_enumerate
+
+
+class IntersectionFinder:
+    """
+    Utility class for finding intersections in a given StatefulTractogram with
+    a diameter for each streamline.
+    """
+
+    FLOAT_EPSILON = 0.0000001
+
+    @property
+    def colliders(self):
+        """Bitmap identifying streamlines that hit another streamline
+        and should be filtered out."""
+        return self._colliders
+
+    @property
+    def collisions(self):
+        """Collision point of each collider."""
+        return self._collisions
+
+    @property
+    def collided(self):
+        """Streamlines hit by colliders. They should not be filtered and
+        are saved simply for visualization."""
+        return self._collided
+
+    @property
+    def excluded(self):
+        """Streamlines that don't collide, but should be excluded for
+        other reasons. (see --tmv_size_threshold)"""
```

Comment:
explain --tmv_size_threshold. We don't know where --tmv_size_threshold is defined either. We should reference a script inside a module because modules can be reused in multiple scripts.